pycalphad/codegen/callables.py

```diff
@@ -67,19 +71,53 @@ def build_callables(dbf, comps, phases, model=None, parameters=None, callables=N
         param_values = np.empty(0)
     comps = sorted(unpack_components(dbf, comps))
     pure_elements = get_pure_elements(dbf, comps)
+    # TODO: Conditions needing Hessians should probably have a 'second-order' tag or something
+    build_hessians = any(str(cond).startswith('MU') for cond in conds.keys())
 
     callables = callables if callables is not None else {}
     _callables = {
         'massfuncs': {},
         'massgradfuncs': {},
+        'masshessfuncs': {},
         'callables': {},
-        'grad_callables': {}
+        'grad_callables': {},
+        'hess_callables': {},
+        'internal_cons': {},
+        'internal_jac': {},
+        'internal_cons_hess': {},
+        'mp_cons': {},
+        'mp_jac': {},
     }
 
     models = unpack_kwarg(model, default_arg=Model)
     param_symbols = [wrap_symbol(sym) for sym in param_symbols]
     phase_records = {}
-    # create models
+
+    state_variables = set()
+    for name in phases:
+        mod = models[name]
+        if isinstance(mod, type):
+            models[name] = mod = mod(dbf, comps, name, parameters=param_symbols)
+        state_variables |= set(mod.state_variables)
+
+    unspecified_statevars = state_variables - set(conds.keys())
+    if len(unspecified_statevars) > 0:
+        #raise ValueError('The following state variables must be specified: {0}'.format(unspecified_statevars))
```

Comment:
For the code as it is today, the correct behavior would be to raise. For the code once NP conditions are supported, the correct behavior may be to pass, or to execute another code path. (Global minimization for the `len(unspecified_statevars) > 0` case is going to be tricky, but that's another issue.)

Since we decided to put NP conditions into a separate PR, I can re-enable the raise here for now.